rcljava/src/main/cpp/org_ros2_rcljava_RCLJava.cpp

```diff
@@ -152,9 +153,30 @@ Java_org_ros2_rcljava_RCLJava_nativeGetRMWIdentifier(JNIEnv * env, jclass)
   return env->NewStringUTF(rmw_implementation_identifier);
 }
 
+#define RCLJAVA_QOS_SET_RMW_TIME(qos_profile, policy_name, seconds, nanos) \
+  do { \
+    if (seconds < 0) { \
+      rcljava_throw_exception( \
+        env, "java/lang/IllegalArgumentException", "negative " #policy_name); \
+      free(qos_profile); \
+      return 0; \
+    } \
+    if (nanos < 0) { \
+      rcljava_throw_exception( \
+        env, "java/lang/IllegalStateException", \
+        "unexpected error: " #policy_name " nanoseconds should not have been negative"); \
```

Comment:
nit: I'd make this exception message look similar to one for seconds

Reply:
Done in https://github.com/osrf/ros2_java/pull/20/commits/f89164180a8470a29ec57a84d28dd940855006f3.